lib/SILOptimizer/Analysis/SimplifyInstruction.cpp

```diff
@@ -752,7 +752,7 @@ swift::replaceAllUsesAndErase(SingleValueInstruction *svi, SILValue newValue,
       callbacks.deleteInst(user);
       continue;
     }
-    use->set(newValue);
+    callbacks.setUseValue(use, newValue);
```

Comment:
This is one of the files where I already knew there were issues with us not having full coverage of IR changes via InstModCallbacks. I went through and fixed it in all of the appropriate places to use callbacks.setUseValue();

Reply:
The obvious fix here would just have been to call the rauw callback, which we'll still need to do anyway.